Added NuGet set api key aliases

devlead opened the pull request:
Added aliases for setting api key for a given source. This allows you to authenticate to private feeds using an API key. 

Example usage:
```csharp
NuGetSetApiKey("3f81e328-3ba1-47eb-a92b-363d804bd132", "http://www.privatenugetfeed.com");
```

patriksvensson commented on src/Cake.Common/Tools/NuGet/SetApiKey/NuGetSetApiKey.cs:
```diff
@@ -0,0 +1,106 @@
+            var builder = new ProcessArgumentBuilder();
+
+            builder.Append("setapikey");
+            builder.AppendQuoted(apiKey);
```
Shouldn't this be `AppendSecret(AppendQuoted(apiKey))`? Otherwise the output will be shown in the build log.

devlead replied:
Agree will fix.

devlead replied:
Agree will fix.

devlead replied:
Will also fix output as nuget outputs
```
The API Key '3f81e328-3ba1-47eb-a92b-363d804bd132' was saved for 'http://www.privatenugetfeed.com'.
```

patriksvensson commented on src/Cake.Common/Tools/NuGet/SetApiKey/NuGetSetApiKey.cs:
```diff
@@ -0,0 +1,106 @@
+namespace Cake.Common.Tools.NuGet.SetApiKey
+{
+    /// <summary>
+    /// The NuGet set API key used to set API key used for API/feed authentication
```
End sentence with period :smile:

devlead replied:
I'll fix.

patriksvensson commented on src/Cake.Common/Tools/NuGet/SetApiKey/NuGetSetApiKey.cs:
```diff
@@ -0,0 +1,106 @@
+        /// Installs NuGet packages using the specified package id and settings.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <param name="source">Server URL where the API key is valid.</param>
```
"THE server URL" :smile:

devlead replied:
Indeed, copy pasted from nuget docs, will fix.

patriksvensson commented on src/Cake.Common/Tools/NuGet/SetApiKey/NuGetSetApiKey.cs:
```diff
@@ -0,0 +1,106 @@
+            return builder;
+        }
+
+         /// <summary>
```
Intendation

patriksvensson commented:
I'm a human StyleCop :wink:

devlead commented:
@patriksvensson this became a rabbit hole :) Found an verbose logging issue outputting "unsafe" values (true for all tools, but could easily fix in this case)
Added so that tool output is validated and added tests for this.

patriksvensson commented:
Merged! :+1:
